File: clients/android/app/src/main/kotlin/my/ssdid/drive/domain/repository/ActivityRepository.kt
package my.ssdid.drive.domain.repository

import my.ssdid.drive.domain.model.FileActivity
import my.ssdid.drive.util.Result

/**
 * Repository interface for activity log operations.
 */
interface ActivityRepository {

    /**
     * Get activity log entries with optional filtering and pagination.
     */
    suspend fun getActivity(
        page: Int? = null,
        pageSize: Int? = null,
        eventType: String? = null,
        resourceType: String? = null
    ): Result<List<FileActivity>>

    /**
     * Get activity log entries for a specific resource.
     */
    suspend fun getResourceActivity(
        resourceId: String,
        page: Int? = null,
        pageSize: Int? = null
    ): Result<List<FileActivity>>
}

File: clients/ios/SsdidDrive/SsdidDrive/Domain/Repository/ActivityRepository.swift
import Foundation

/// Repository protocol for file activity log operations
protocol ActivityRepository {

    /// Fetch activity logs with optional filtering
    /// - Parameters:
    ///   - page: Page number (1-based)
    ///   - pageSize: Number of items per page
    ///   - eventType: Optional event type filter (e.g. "file.uploaded")
    /// - Returns: Paginated activity response
    func getActivity(page: Int, pageSize: Int, eventType: String?) async throws -> ActivityResponse

    /// Fetch activity logs for a specific resource
    /// - Parameters:
    ///   - resourceId: ID of the file or folder
    ///   - page: Page number (1-based)
    ///   - pageSize: Number of items per page
    /// - Returns: Paginated activity response
    func getResourceActivity(resourceId: String, page: Int, pageSize: Int) async throws -> ActivityResponse
}
